python/ray/tune/utils/file_transfer.py

```diff
@@ -0,0 +1,236 @@
+import io
+import os
+import shutil
+import tarfile
+
+from typing import Optional, Tuple, Dict, Generator, Union
+
+import ray
+from ray.exceptions import RayTaskError
+
+
+DEFAULT_CHUNK_SIZE = 500 * 1024 * 1024
+
+
+def sync_dir_between_nodes(
+    source_ip: str,
+    source_path: str,
+    target_ip: str,
+    target_path: str,
+    force_all: bool = False,
+    chunk_size: int = DEFAULT_CHUNK_SIZE,
+    _return_all_remotes: bool = False,
```

Comment:
Even though this is nonpublic, is it worth documenting what this does for maintainability purposes?

Reply:
Added